src/player.cpp

```diff
@@ -4811,8 +4817,9 @@ void player::update_needs( int rate_multiplier )
 
     // Note: intentionally not in metabolic rate
     if( has_recycler ) {
-        hunger_rate *= 0.5f;
-        thirst_rate = std::min( thirst_rate, std::max( 0.5f, thirst_rate * 0.5f ) );
+        // Recycler won't help much with mutant metabolism - it is indended for human one
+        hunger_rate = std::min( hunger_rate, std::max( 0.5f, hunger_rate - 0.5f ) );
+        thirst_rate = std::min( thirst_rate, std::max( 0.5f, thirst_rate - 0.5f ) );
```

Comment:
The problem is, the maximum value changes too.
Clamp expects `min <= max`, which can't be guaranteed here: `hunger_rate - 0.5` can be above or below `0.5`.